tracer/src/Datadog.Trace/Ci/TestModule.cs

```diff
@@ -0,0 +1,315 @@
+            CIWorkspacePath = environment.WorkspacePath,
+            GitRepository = environment.Repository,
+            GitCommit = environment.Commit,
+            GitBranch = environment.Branch,
+            GitTag = environment.Tag,
+            GitCommitAuthorDate = environment.AuthorDate?.ToString("yyyy-MM-dd'T'HH:mm:ss.fffK", CultureInfo.InvariantCulture),
+            GitCommitAuthorName = environment.AuthorName,
+            GitCommitAuthorEmail = environment.AuthorEmail,
+            GitCommitCommitterDate = environment.CommitterDate?.ToString("yyyy-MM-dd'T'HH:mm:ss.fffK", CultureInfo.InvariantCulture),
+            GitCommitCommitterName = environment.CommitterName,
+            GitCommitCommitterEmail = environment.CommitterEmail,
+            GitCommitMessage = environment.Message,
+            BuildSourceRoot = environment.SourceRoot,
+            LibraryVersion = TracerConstants.AssemblyVersion,
+            RuntimeName = frameworkDescription.Name,
+            RuntimeVersion = frameworkDescription.ProductVersion,
+            RuntimeArchitecture = frameworkDescription.ProcessArchitecture,
+            OSArchitecture = frameworkDescription.OSArchitecture,
+            OSPlatform = frameworkDescription.OSPlatform,
+            OSVersion = Environment.OSVersion.VersionString,
+        };
+
+        if (environment.VariablesToBypass is { } variablesToBypass)
+        {
+            tags.CiEnvVars = Vendors.Newtonsoft.Json.JsonConvert.SerializeObject(variablesToBypass);
+        }
+
+        // Check if Intelligent Test Runner has skippable tests
+        if (CIVisibility.HasSkippableTests())
+        {
+            tags.TestsSkipped = "true";
+        }
+
+        var span = Tracer.Instance.StartSpan(
+            string.IsNullOrEmpty(framework) ? "test_module" : $"{framework!.ToLowerInvariant()}.test_module",
+            tags: tags,
+            startTime: startDate);
+
+        span.Type = SpanTypes.TestModule;
+        span.ResourceName = name;
+        span.SetTag(Trace.Tags.Origin, TestTags.CIAppTestOriginName);
+        span.SetTraceSamplingPriority(SamplingPriority.AutoKeep);
+
+        tags.ModuleId = span.SpanId;
+
+        _span = span;
+        Current = this;
+        CIVisibility.Log.Debug("### Test Module Created: {name}", name);
+
+        if (startDate is null)
+        {
+            // If a module doesn't have a fixed start time we reset it before running code
+            span.ResetStartTime();
+        }
+    }
+
+    /// <summary>
+    /// Gets the current TestModule
+    /// </summary>
+    public static TestModule? Current
+    {
+        get => CurrentModule.Value;
+        internal set => CurrentModule.Value = value;
+    }
+
+    /// <summary>
+    /// Gets the module name
+    /// </summary>
+    public string Name { get; }
+
+    /// <summary>
+    /// Gets the test module start date
+    /// </summary>
+    public DateTimeOffset StartDate => _span.StartTime;
```

Comment:
Can we rename this to `StartTime`?

Reply:
done in [df65ebf](https://github.com/DataDog/dd-trace-dotnet/pull/3276/commits/df65ebf1281d96e00d474e1aa0eb34300493b830)